build-support/bin/bootstrap_pants_pex.sh

```diff
@@ -5,10 +5,12 @@ cd "$REPO_ROOT" || exit 1
 
 # This script is used to generate pants.pex, which is used to run Pants in CI.
 
-export PY="${PY:-python3}"
-
 # shellcheck source=build-support/common.sh
 source "${REPO_ROOT}/build-support/common.sh"
+
+PY="$(determine_python)"
+export PY
```

Comment:
I spent way too much time trying to debug why `export PY="$(determine_python)"` was failing, until I remembered the wonders of shellcheck.